src/Pyro4/core.py

```diff
@@ -1481,6 +1481,17 @@ def uriFor(self, objectOrId, nat=True):
             loc = self.locationStr
         return URI("PYRO:%s@%s" % (objectOrId, loc))
 
+    def resetMetadataCache(self, objectOrId, nat=True):
+        '''Reset cache of metadata when a Daemon has available methods/attributes
+        dynamically updated.  Clients will have to get a new proxy to see changes'''
+        uri = self.uriFor(objectOrId, nat)
+        # can only be cached if registered, else no-op
+        if uri.objectOrId in self.objectsById:
+            registered_object = self.objectsById[uri.object]
+            # Clear cache regardless of how it is accessed (as_lists seems to be used inconsistently)
+            util.reset_exposed_members(registered_object, Pyro4.config.REQUIRE_EXPOSE, as_lists=True)
+            util.reset_exposed_members(registered_object, Pyro4.config.REQUIRE_EXPOSE, as_lists=False)
+

```

Comment:
It seems wasteful to re-scan exposed members twice (once for sets once for lists) but I guess we'll not be calling this reset method often anyway